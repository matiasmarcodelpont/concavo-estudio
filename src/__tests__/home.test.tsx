import '@testing-library/jest-dom'
import { render, screen, within } from '@testing-library/react'

import Home from '../app/page'

jest.mock('../data/data.json', () => ({
  products: [
    {
      slug: 'azulejos-los-peores',
      name: 'Azulejos los peores',
      contributor: {
        slug: 'facu',
      },
    },
    {
      slug: 'azulejos-verdes',
      name: 'Azulejos verdes',
      contributor: null,
    },
    {
      slug: 'azulejos-los-mejores',
      name: 'Azulejos los mejores',
      contributor: {
        slug: 'mati',
      },
    },
    {
      slug: 'azulejos-azules',
      name: 'Azulejos azules',
      contributor: null,
    },
  ],
}))

describe('Home', () => {
  it('renders homepage unchanged', () => {
    const { container } = render(<Home />)
    expect(container).toMatchSnapshot()
  })

  it('renders concavo products', () => {
    render(<Home />)

    const productsList = screen.getByRole('list')
    expect(productsList).toBeInTheDocument()

    const products = within(productsList).getAllByRole('listitem')
    expect(products).toHaveLength(2)

    expect(products[0]).toBeInTheDocument()
    const link0 = within(products[0]).getByRole('link', { name: 'Azulejos verdes' }) as HTMLAnchorElement
    expect(link0).toBeInTheDocument()
    expect(link0.href).toMatch(/^https?:\/\/[^/]+\/productos\/azulejos-verdes$/)

    expect(products[1]).toBeInTheDocument()
    const link1 = within(products[1]).getByRole('link', { name: 'Azulejos azules' }) as HTMLAnchorElement
    expect(link1).toBeInTheDocument()
    expect(link1.href).toMatch(/^https?:\/\/[^/]+\/productos\/azulejos-azules$/)
  })
})
