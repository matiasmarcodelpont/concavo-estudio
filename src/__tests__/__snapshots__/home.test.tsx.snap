// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
<div>
  <main>
    <ul>
      <li>
        <a
          href="/productos/azulejos-verdes"
        >
          Azulejos verdes
        </a>
      </li>
      <li>
        <a
          href="/productos/azulejos-azules"
        >
          Azulejos azules
        </a>
      </li>
    </ul>
  </main>
</div>
`;

exports[`Page renders homepage unchanged 1`] = `
<div>
  <main>
    <ul>
      <li>
        <a
          href="/productos/azulejos-verdes"
        >
          Azulejos verdes
        </a>
      </li>
      <li>
        <a
          href="/productos/azulejos-azules"
        >
          Azulejos azules
        </a>
      </li>
    </ul>
  </main>
</div>
`;
